packages/dui3/components/model/Receiver.vue

```diff
@@ -0,0 +1,255 @@
+import { versionDetailsQuery } from '~/lib/graphql/mutationsAndQueries'
+import type { VersionListItemFragment } from '~/lib/common/generated/gql/graphql'
+import { useMixpanel } from '~/lib/core/composables/mixpanel'
+import { useInterval, watchOnce } from '@vueuse/core'
+
+const { trackEvent } = useMixpanel()
+const app = useNuxtApp()
+
+const props = defineProps<{
+  modelCard: IReceiverModelCard
+  project: ProjectModelGroup
+}>()
+
+const store = useHostAppStore()
+
+const openVersionsDialog = ref(false)
+
+app.$baseBinding.on('documentChanged', () => {
+  openVersionsDialog.value = false
+})
+
+const isExpired = computed(() => {
+  return props.modelCard.latestVersionId !== props.modelCard.selectedVersionId
+})
+
+// Cancels any in progress receive AND load selected version
+const handleVersionSelection = async (
+  selectedVersion: VersionListItemFragment,
+  latestVersion: VersionListItemFragment
+) => {
+  openVersionsDialog.value = false
+  void trackEvent('DUI3 Action', {
+    name: 'Load Card Version Change',
+    isLatestVersion: selectedVersion === latestVersion
+  })
+  if (props.modelCard.progress) {
+    await store.receiveModelCancel(props.modelCard.modelCardId)
+  }
+  await store.patchModel(props.modelCard.modelCardId, {
+    selectedVersionId: selectedVersion.id,
+    latestVersionId: latestVersion.id, // patch this dude as well, to make sure
+    hasSelectedOldVersion: selectedVersion.id === latestVersion.id
+  })
+
+  await store.receiveModel(props.modelCard.modelCardId)
+}
+
+// Cancels any in progress receive OR receives latest version
+const handleMainButtonClick = async () => {
+  if (props.modelCard.progress)
+    return await store.receiveModelCancel(props.modelCard.modelCardId)
+  await receiveLatestVersion()
+}
+
+// Cancels any in progress receive AND receives latest version
+const receiveLatestVersion = async () => {
+  // Note: here we're updating the model card info, and afterwards we're hitting the receive action
+  await store.patchModel(props.modelCard.modelCardId, {
+    selectedVersionId: props.modelCard.latestVersionId
+  })
+  if (props.modelCard.progress)
+    await store.receiveModelCancel(props.modelCard.modelCardId)
+  await store.receiveModel(props.modelCard.modelCardId)
+}
+
+const expiredNotification = computed(() => {
+  if (!props.modelCard.latestVersionId || props.modelCard.hasDismissedUpdateWarning)
+    return
+  if (props.modelCard.latestVersionId === props.modelCard.selectedVersionId) return
+  const notification = {} as ModelCardNotification
+  notification.dismissible = true
+  notification.level = 'warning'
+  notification.text = 'Newer version available!'
+  notification.cta = {
+    name: 'Update',
+    action: receiveLatestVersion
+  }
+  return notification
+})
+
+const receiveResultNotification = computed(() => {
+  if (
+    !props.modelCard.bakedObjectIds ||
+    props.modelCard.displayReceiveComplete !== true
+  )
+    return
+
+  const notification = {} as ModelCardNotification
```

Comment:
Bad practice, cause at this point `notification` is not a ModelCardNotification. If you end up adding another required prop to that type, you will not get any errors here, because you're telling TS that on line 177 the object is already correct w/ all required properties.

Do this instead, to ensure all required props are set:
```
const notification: ModelCardNotification = {
  dismissable: true,
  report: props.modelCard.report,
   ...
}
```